models/demos/bert/tt/ttnn_optimized_bert.py

```diff
@@ -42,7 +42,7 @@ def bert_attention(
     attention_scores = ttnn.matmul(
         query,
         key,
-        memory_config=ttnn.L1_MEMORY_CONFIG,
+        memory_config=ttnn.DRAM_MEMORY_CONFIG if getattr(config, "use_dram", False) else ttnn.L1_MEMORY_CONFIG,
         dtype=ttnn.bfloat16,
         core_grid=ttnn.CoreGrid(y=batch_size, x=num_cores_x),
```

Comment:
I see num_cores_x=12 by default, which would be wrong for WH. Can you check the coregrid size?

Reply:
I will check and update core_grid size for WH.

Reply:
I've updated.

Reply:
Do CI tests pass after this change? We might need to re-run the tests.

Reply:
Yes, I have the updated the PR [description](https://github.com/tenstorrent/tt-metal/pull/8896#issue-2320907336) with CI links.

Reply:
@uaydonat, 
Having attention_scores on L1 regardless of the core_grid, I encounter this error: `Statically allocated circular buffers in program 9 clash with L1 buffers on core range [(x=0,y=0) - (x=7,y=6)]. L1 buffer allocated at 342912 and static circular buffer region ends at 445952`. 

Switching the dtype to bf8_b helped here.

- PCC for the model using bf8_b dtype attention_scores compared to bf16 attention_scores :
   - Remained the same on both GS (0.89) and WH (0.94).
- PCC for the QA task using bf8_b dtype attention_scores compared to bf16 attention_scores::
  - Start Logits:
     - GS: Stayed constant at 0.81.
     - WH: Improved slightly from 0.89 to 0.90.
  - End Logits:
     - GS: Dropped slightly from 0.80 to 0.79.
     - WH: Increased from 0.87 to 0.90.

 Is it feasible to proceed with bf8_b here?

Reply:
Does not look like pcc changed drastically, so it is ok.